src/sentry/monitors/validators.py

```diff
@@ -198,11 +198,11 @@ class MonitorValidator(CamelSnakeSerializer):
     project = ProjectField(scope="project:read")
     name = serializers.CharField(max_length=128)
     slug = serializers.RegexField(
-        r"^[a-zA-Z0-9_-]+$",
+        DEFAULT_SLUG_PATTERN,
```

Comment:
Same as previous PRs this will be addressed and won't need flag if you do this: https://github.com/getsentry/sentry/pull/55122#discussion_r1300521992